src/goto-programs/loop_ids.h

```diff
@@ -31,14 +31,28 @@ struct loop_idt
   {
   }
 
+  loop_idt() : function_id(""), loop_number()
+  {
+  }
+
```

Comment:
When we compute the cause loop id, the loop number without value indicates that no cause loop found. This constructor constructs a loop id with `loop_number` having no value.

https://github.com/diffblue/cbmc/blob/f5629d636c7e9b2db4f7e8f8851b74bca12b4fce/src/goto-synthesizer/cegis_verifier.cpp#L480

Reply:
Hmm, yes, so I think this reinforces [the comment](https://github.com/diffblue/cbmc/pull/7430#discussion_r1048112384) that I provided in that function: I believe that function should return an `optionalt<loop_idt>` rather than (possibly) returning a not-really-initialised `loop_idt`.